### PR TITLE
Smddp MNIST example update

### PR DESCRIPTION
**Pull request by apoorvtintin**
*Issue #, if available:*

*Description of changes:*
Updated SMDDP MNIST training example with new APIs and information

*Testing done:*
Yes, tested on Sagemaker

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [x] I have tested my notebook(s) and ensured it runs end-to-end
- [x] I have linted my notebook(s) and code using `tox -e black-format,black-nb-format`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by Zha0q1** [comment]
@apoorvtintin Thanks for the PR! Code changes look good to me!
Would you help update the notebook too? There's some outdated info e.g.
1. "This notebook example shows how to use smdistributed.dataparallel with PyTorch in SageMaker using MNIST dataset." (introduce smddp backend explicitly)
2. "The [data parallel](https://docs.aws.amazon.com/sagemaker/latest/dg/data-parallel.html) feature in this library is a distributed data parallel training framework for PyTorch, TensorFlow, and MXNet. (remove mxnet)"
3. "The training script is very similar to a PyTorch training script you might run outside of SageMaker, but modified to run with the smdistributed.dataparallel library. This library's PyTorch client provides an alternative to PyTorch's native DDP." (we DO use pytorch native ddp now)
etc.

**Review by Zha0q1 — CHANGES_REQUESTED** [review]
@apoorvtintin Thanks for the PR! Code changes look good to me!
Would you help update the notebook too? There's some outdated info e.g.
1. "This notebook example shows how to use smdistributed.dataparallel with PyTorch in SageMaker using MNIST dataset." (introduce smddp backend explicitly)
2. "The [data parallel](https://docs.aws.amazon.com/sagemaker/latest/dg/data-parallel.html) feature in this library is a distributed data parallel training framework for PyTorch, TensorFlow, and MXNet. (remove mxnet)"
3. "The training script is very similar to a PyTorch training script you might run outside of SageMaker, but modified to run with the smdistributed.dataparallel library. This library's PyTorch client provides an alternative to PyTorch's native DDP." (we DO use pytorch native ddp now)
etc.

**Review by mchoi8739 — DISMISSED** [review]

**Comment by apoorvtintin** [comment]
> @apoorvtintin Thanks for the PR! Code changes look good to me! Would you help update the notebook too? There's some outdated info e.g.
> 
>     1. "This notebook example shows how to use smdistributed.dataparallel with PyTorch in SageMaker using MNIST dataset." (introduce smddp backend explicitly)
> 
>     2. "The [data parallel](https://docs.aws.amazon.com/sagemaker/latest/dg/data-parallel.html) feature in this library is a distributed data parallel training framework for PyTorch, TensorFlow, and MXNet. (remove mxnet)"
> 
>     3. "The training script is very similar to a PyTorch training script you might run outside of SageMaker, but modified to run with the smdistributed.dataparallel library. This library's PyTorch client provides an alternative to PyTorch's native DDP." (we DO use pytorch native ddp now)
>        etc.

Thanks for the review and comments Zhaoqi and Miyoung, I have updated the vocabulary for the notebook to reflect our new smddp backend. 
Please let me know if any more changes are needed.

File: training/distributed_training/pytorch/data_parallel/mnist/pytorch_smdataparallel_mnist_demo.ipynb [hunk 1]
```diff
@@ -7,9 +7,9 @@
     "# Part 1: Distributed data parallel MNIST training with PyTorch and SageMaker distributed\n",
```
Comment (Zha0q1) [inline training/distributed_training/pytorch/data_parallel/mnist/pytorch_smdataparallel_mnist_demo.ipynb:7]: <p>nit: "PyTorch distributed library" -&gt; "PyTorch DDP (Distributed Data Parallel)"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3448/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (Zha0q1) [inline training/distributed_training/pytorch/data_parallel/mnist/pytorch_smdataparallel_mnist_demo.ipynb:7]: <p>similarly "PyTorch DDP"</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3448/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Review by Zha0q1 — APPROVED** [review]
LGTM, thank you!

File: training/distributed_training/pytorch/data_parallel/mnist/pytorch_smdataparallel_mnist_demo.ipynb (hunk 1 above)
Comment (jkroll-aws) [inline training/distributed_training/pytorch/data_parallel/mnist/pytorch_smdataparallel_mnist_demo.ipynb:7]: <p>Nit: <code>Pytorch</code> -&gt; <code>PyTorch</code>   </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3448/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Comment by zhaoqizqwang** [comment]
Closing this PR since it was created a long time ago and didn't pass all the checks. Feel free to open a new pull request instead.

**Pull request closed without merging**